apps/cow-tools/src/app/milkman/(components)/TransactionCard.tsx

```diff
@@ -0,0 +1,90 @@
+import { Address, NetworkChainId, networkFor } from "@bleu-fi/utils";
+import { useState } from "react";
+import { FieldValues } from "react-hook-form";
+
+import { stages } from "#/app/milkman/(components)/TransactionProgressBar";
+import { TransactionStatus } from "#/app/milkman/utils/type";
+import { useOrder } from "#/contexts/OrderContext";
+
+import { FormHeader } from "./orderForm/Header";
+import { FormOrderOverview } from "./orderForm/Overview";
+import { FormSelectPriceChecker } from "./orderForm/PriceChecker";
+import { OrderResume } from "./orderForm/Resume";
+
+export function TransactionCard({
+  userAddress,
+  chainId,
+}: {
+  userAddress: Address;
+  chainId?: NetworkChainId;
+  walletAmount?: string;
+  walletAmountBigNumber?: bigint;
+}) {
+  const { transactionStatus, setTransactionStatus } = useOrder();
+  const network = networkFor(chainId);
+
+  const [orderOverviewData, setOrderOverviewData] = useState<FieldValues>();
+  const [priceCheckerData, setPriceCheckerData] = useState<FieldValues>();
+
+  function handleBack() {
+    const currentStage = stages.find(
+      (stage) => stage.name === transactionStatus,
+    );
+    setTransactionStatus(currentStage?.previousStage ?? transactionStatus);
+  }
+
+  function handleContinue() {
+    const currentStage = stages.find(
+      (stage) => stage.name === transactionStatus,
+    );
+    setTransactionStatus(currentStage?.nextStage ?? transactionStatus);
+  }
+
+  const FORM_CONTENTS: { [key in TransactionStatus]?: JSX.Element } = {
+    [TransactionStatus.ORDER_OVERVIEW]: (
+      <FormOrderOverview
+        onSubmit={(data: FieldValues) => {
+          setOrderOverviewData(data);
+          handleContinue();
+        }}
+        userAddress={userAddress}
+        defaultValues={orderOverviewData}
+      />
+    ),
+    [TransactionStatus.ORDER_STRATEGY]: (
+      <FormSelectPriceChecker
+        onSubmit={(data: FieldValues) => {
+          setPriceCheckerData(data);
+          handleContinue();
+        }}
+        defaultValues={priceCheckerData}
+        tokenSellAddress={orderOverviewData?.tokenSell.address}
+        tokenBuyAddress={orderOverviewData?.tokenBuy.address}
+      />
+    ),
+    [TransactionStatus.ORDER_RESUME]: (
```

Comment:
I think you mean this? ```suggestion
    [TransactionStatus.ORDER_SUMMARY]: (
```